CONTRIBUTING.md

```diff
@@ -14,6 +14,23 @@ Please note we have a code of conduct, please follow it in all your interactions
 4. Wait for the maintainers to review your pull request and do the changes if requested.
 5. *We use danger, so update go to https://travis-ci.org and login with github. Next you will see a `+` near my repositories. Click on it and select your fork enabling it for Travis. Then go the dashboard and add two environment varibales which two wont get displayed in the terminal(There's an option). The first one must be named `USERNAME` and should contain your github username. The second one should be named `API_TOKEN` and should contain a github personal token with atleast scope to public repos. It enables fast and easy moderation adequate testing.*
 
+## Contributions Best Practices
+
+**Commits**
+* Write clear meaningful git commit messages (Do read http://chris.beams.io/posts/git-commit/)
+* Make sure your PR's description contains GitHub's special keyword references that automatically close the related issue when the PR is merged. (More info at https://github.com/blog/1506-closing-issues-via-pull-requests )
+* When you make very very minor changes to a PR of yours (like for example fixing a failing travis build or some small style corrections or minor changes requested by reviewers) make sure you squash your commits afterwards so that you don't have an absurd number of commits for a very small fix. (Learn how to squash at https://davidwalsh.name/squash-commits-git )
```

Comment:
Very minor.

Reply:
I think these are Important for a Beginner to Contribute.